internal/clients/service.go

```diff
@@ -22,28 +32,61 @@ func NewTemporalService(configData []byte) (*TemporalServiceImpl, error) {
 	var conf = TemporalServiceConfig{}
 	err := json.Unmarshal(configData, &conf)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to unmarshal config data: %w", err)
 	}
 
 	logger := slog.New(slog.NewJSONHandler(os.Stdout, &slog.HandlerOptions{
 		AddSource: true,
 		Level:     slog.LevelDebug,
 	}))
 
+	logger.Debug("Starting NewTemporalService", slog.String("hostPort", conf.HostPort), slog.Bool("useTLS", conf.UseTLS))
+
+	var dialOptions []grpc.DialOption
+	if conf.UseTLS && conf.CACert != "" && conf.CertFile != "" && conf.KeyFile != "" {
```

Comment:
If `UseTLS` is `true`, we should error if necessary config values are empty ( `conf.CACert != "" && conf.CertFile != "" && conf.KeyFile != ""`) and not fallback to nonTLS